### DISCUSSION TITLE
Methods displayed at checkout

### DISCUSSION THREAD

**Discussion by euluizaraujo** [juspay/hyperswitch#3028] [Q&A] [answered]
Can I create an option in the routing so that wallets such as Google Pay and Apple Pay are only available to customers from a certain country? If so, is this automatically reflected at checkout?

Example:

For a purchase from Spain over US$100, process via stripe but only with Apple Pay, so at checkout just use the Apple Pay button;

For a purchase from Mexico over US$100, process via Stripe but with Google Play and credit card and consequently these methods automatically appear at checkout and it does not look like Apple Pay.

**Comment by euluizaraujo** [comment]
In summary, I would like to create conditions to route by payment method, is it possible?

**Comment by euluizaraujo** [comment]
Another question related to routing, after creating a routing rule I can no longer edit or delete it? This seems like a problem to me.

**Comment by vspecky** [comment, marked as answer]
@euluizaraujo Hey! yes, we support configuring constraints like these through the connector account APIs. These are part of your connector account integration as opposed to routing. And all the payment methods displayed to your customers are validated against the provided constraints. You can view all the supported parameters in the [api reference](https://api-reference.hyperswitch.io/api-reference/merchant-connector-account/merchant-connector--create).
![image](https://github.com/juspay/hyperswitch/assets/47862918/4176cb22-2360-4998-b84b-4d72543bf464)

**Reply by manojradhakrishnan** [reply]
@tinu-hareesswar Let's please include this FAQ on hyperswitch-docs

**Comment by euluizaraujo** [comment]
thanks @vspecky. In this case, do I need to configure country or currency rules via API when activating a connector and does this reflect in routing? Via Panel still not possible?

Does this setting automatically reflect in unified checkout? Ex: I configured the PayPal wallet to be activated only for Mexico, do I need to change something at checkout or is this already reflected in it?

**Reply by vspecky** [reply]
@euluizaraujo Yes, currently this feature is only available in the API, but we'll be adding Dashboard support for configuring it soon. Also you don't require changes in unified checkout for any changes to take place.

**Reply by euluizaraujo** [reply]
@vspecky Will this configuration via dashboard be added to the routing configuration or connector configuration area?

**Reply by vspecky** [reply]
@euluizaraujo This will be accessible in the connector configuration area. In general, Routing is in charge of choosing a connector. If further refinements are required for a connector in terms of constraining enabled payment methods based on parameters like customer country, that will be part of the connector configuration.
